fix(marine_ops): _generate_chain_database diameters[:20] excludes 76mm — clears 4 of 77 marine_ops failures

Sub-issue of [vamseeachanta/workspace-hub#2609](https://github.com/vamseeachanta/workspace-hub/issues/2609) cluster R4. 1-line test/fixture bug, ~4 test failures.

## Symptom

4 failing tests in `tests/marine_ops/marine_engineering/legacy/test_component_database.py`:

- `test_get_chain_by_specification`
- `test_find_chain_with_safety_factor`
- `test_database_performance`
- `test_weight_scales_with_diameter_squared`

Top error:
```
tests/marine_ops/marine_engineering/legacy/test_component_database.py:589: in test_get_chain_by_specification
    chain = database.get_chain(...)
tests/marine_ops/marine_engineering/legacy/test_component_database.py:328: in get_chain
    raise ValueError(
E   ValueError: Chain not found: 76mm R4 Stud Link
```

## Root cause

`tests/marine_ops/marine_engineering/legacy/test_component_database.py:236` (verified at HEAD `08e9c33`):

```python
for diameter in diameters[:20]:  # Generate 60 components
```

The `diameters` array is declared on lines 228–231:
```
[16, 19, 22, 24, 28, 32, 34, 36, 38, 40, 44, 46, 48, 50, 52,
 56, 58, 60, 62, 64, 66, 68, 70, 76, 78, 81, 84, 87, 90, 92, ...]
```

76mm sits at index 23, so `diameters[:20]` truncates at 64mm and the four tests querying `76mm R4 Stud Link` cannot find their fixture.

A second copy of the same bug exists at `tests/marine_ops/marine_engineering/test_component_database.py:236` (non-legacy path) — the fix should sweep both unless the divergence is intentional.

## Fix shape

~1–2 LoC + test-data array re-verification. Change `diameters[:20]` to either:

- **(a) The full list**: `for diameter in diameters:` — generates all 49 diameters × 3 grades = 147 components. The existing assertion on line 219 (`Loaded 60 chain components` print, plus any `len(...) == 60` test assertion) will need its expected count updated.
- **(b) A slice that includes 76mm**: e.g., `diameters[:24]` — generates 24 × 3 = 72 components, again breaking the "60 components" expectation.

### Sub-decision (please confirm before implementation)

Was the test author's intent:

1. **"First 20 standard sizes"** — then the array order is wrong (76mm should be among the first 20) and the array needs reordering, slice stays `[:20]`.
2. **"All standard sizes up to and including 76mm"** — slice extends to `[:24]` and the `60` row-count expectation is replaced with `72`.
3. **"All standard sizes"** — drop the slice entirely and replace `60` with `147`.

Reading the comment `# Generate 60 components` and the `Loaded 60 chain components` print on line 219, intent (1) is the most plausible original design but is incompatible with current test queries for 76mm. Intent (2) or (3) appears to be what the test queries actually require.

## Acceptance

- 4 named tests above pass
- No regression in other `test_component_database.py` tests (legacy + non-legacy paths)
- The "Loaded N chain components" print and any related `len(chain_db)` assertions are updated consistently

## Estimated diff

~1–2 LoC in the slice + 1–2 LoC in the count assertion(s). Test-data array re-verification for ordering if intent (1) is chosen.

## References

- Triage plan: `docs/plans/2026-05-03-2609-marine-ops-triage.md` (R4, lines 166–187)
- Parent: [vamseeachanta/workspace-hub#2609](https://github.com/vamseeachanta/workspace-hub/issues/2609)
- Predecessor: [vamseeachanta/workspace-hub#2580](https://github.com/vamseeachanta/workspace-hub/issues/2580) (closed)
- Bug locations at HEAD `08e9c33`:
  - `tests/marine_ops/marine_engineering/legacy/test_component_database.py:236`
  - `tests/marine_ops/marine_engineering/test_component_database.py:236`

--- a/tests/marine_ops/marine_engineering/legacy/test_component_database.py
+++ b/tests/marine_ops/marine_engineering/legacy/test_component_database.py
@@ -221,7 +221,7 @@ class ComponentDatabase:
         print(f"Loaded {len(self.line_db)} synthetic line components")
 
     def _generate_chain_database(self) -> pd.DataFrame:
-        """Generate chain database (60 components)."""
+        """Generate chain database (72 components)."""
         data = []
 
         # Standard diameters (16mm to 162mm)
@@ -233,7 +233,7 @@ class ComponentDatabase:
         grades = [ChainGrade.R3, ChainGrade.R4, ChainGrade.R5]
         link_types = [LinkType.STUD_LINK]
 
-        for diameter in diameters[:20]:  # Generate 60 components
+        for diameter in diameters[:24]:  # Generate 72 components (24 diameters × 3 grades, includes 76mm)
             for grade in grades:
                 chain = ChainProperties.from_excel_formula(diameter, grade, link_types[0])
                 data.append({
@@ -562,10 +562,10 @@ class TestComponentDatabase:
         return ComponentDatabase()
 
     def test_database_loading(self, database):
-        """Validate all 336 components load correctly."""
-        # Chain: 60 components (20 diameters × 3 grades)
-        assert len(database.chain_db) == 60, (
-            f"Expected 60 chain components, got {len(database.chain_db)}"
+        """Validate all 348 components load correctly."""
+        # Chain: 72 components (24 diameters × 3 grades)
+        assert len(database.chain_db) == 72, (
+            f"Expected 72 chain components, got {len(database.chain_db)}"
         )
 
         # Wire: 24 components (8 diameters × 3 constructions)
@@ -578,10 +578,10 @@ class TestComponentDatabase:
             f"Expected 252 synthetic components, got {len(database.line_db)}"
         )
 
-        # Total: 336 components
+        # Total: 348 components
         total = len(database.chain_db) + len(database.wire_db) + len(database.line_db)
-        assert total == 336, (
-            f"Expected 336 total components, got {total}"
+        assert total == 348, (
+            f"Expected 348 total components, got {total}"
         )
 
     def test_get_chain_by_specification(self, database):

--- a/tests/marine_ops/marine_engineering/test_component_database.py
+++ b/tests/marine_ops/marine_engineering/test_component_database.py
@@ -221,7 +221,7 @@ class ComponentDatabase:
         print(f"Loaded {len(self.line_db)} synthetic line components")
 
     def _generate_chain_database(self) -> pd.DataFrame:
-        """Generate chain database (60 components)."""
+        """Generate chain database (72 components)."""
         data = []
 
         # Standard diameters (16mm to 162mm)
@@ -233,7 +233,7 @@ class ComponentDatabase:
         grades = [ChainGrade.R3, ChainGrade.R4, ChainGrade.R5]
         link_types = [LinkType.STUD_LINK]
 
-        for diameter in diameters[:20]:  # Generate 60 components
+        for diameter in diameters[:24]:  # Generate 72 components (24 diameters × 3 grades, includes 76mm)
             for grade in grades:
                 chain = ChainProperties.from_excel_formula(diameter, grade, link_types[0])
                 data.append({
@@ -562,10 +562,10 @@ class TestComponentDatabase:
         return ComponentDatabase()
 
     def test_database_loading(self, database):
-        """Validate all 336 components load correctly."""
-        # Chain: 60 components (20 diameters × 3 grades)
-        assert len(database.chain_db) == 60, (
-            f"Expected 60 chain components, got {len(database.chain_db)}"
+        """Validate all 348 components load correctly."""
+        # Chain: 72 components (24 diameters × 3 grades)
+        assert len(database.chain_db) == 72, (
+            f"Expected 72 chain components, got {len(database.chain_db)}"
         )
 
         # Wire: 24 components (8 diameters × 3 constructions)
@@ -578,10 +578,10 @@ class TestComponentDatabase:
             f"Expected 252 synthetic components, got {len(database.line_db)}"
         )
 
-        # Total: 336 components
+        # Total: 348 components
         total = len(database.chain_db) + len(database.wire_db) + len(database.line_db)
-        assert total == 336, (
-            f"Expected 336 total components, got {total}"
+        assert total == 348, (
+            f"Expected 348 total components, got {total}"
         )
 
     def test_get_chain_by_specification(self, database):

